pkg/controller/master/image/vm_image_controller.go

```diff
@@ -60,31 +60,38 @@ func (h *vmImageHandler) OnChanged(_ string, image *harvesterv1.VirtualMachineIm
 		return image, nil
 	}
 
-	needRetry := false
 	bi, err := util.GetBackingImage(h.backingImageCache, image)
 	if err != nil {
 		if !errors.IsNotFound(err) {
-			logrus.Errorf("%v/%v failed to get backing image, err: %v", image.Namespace, image.Name, err)
+			logrus.WithError(err).WithFields(logrus.Fields{
+				"namespace": image.Namespace,
+				"name":      image.Name,
+			}).Error("failed to get backing image for vmimage")
 			return image, err
 		}
-		needRetry = true
-	} else if bi.DeletionTimestamp != nil {
+		return h.handleRetry(image)
+	}
+
+	if bi.DeletionTimestamp != nil {
 		h.imageController.EnqueueAfter(image.Namespace, image.Name, checkBackingImageInterval)
 		return image, nil
-	} else {
-		for _, status := range bi.Status.DiskFileStatusMap {
-			if status.State == lhv1beta2.BackingImageStateFailed {
-				needRetry = true
-				break
-			}
-		}
 	}
 
-	if needRetry {
-		if image.Status.LastFailedTime == "" {
-			return h.initialize(image)
+	for _, status := range bi.Status.DiskFileStatusMap {
+		if status.State == lhv1beta2.BackingImageStateFailed {
+			return h.handleRetry(image)
 		}
+	}
 
+	_, scErr := util.GetStorageClass(h.storageClassCache, image)
```

Comment:
L75 checks `bi.DeletionTimestamp != nil`, should  L97 do same?

Reply:
I thought you mean L87,  you are right, we need to do the same thing .

Reply:
I've addressed the comment in 8cc0ea0, please take another look. Thanks ! :smiley: